tensornetwork/matrixproductstates/mps.py

```diff
@@ -341,6 +431,315 @@ def measure_two_body_correlator(self, op1: Union[BaseNode, Tensor],
+
+  def get_node(self, site: int) -> BaseNode:
+    """
+    Returns the `Node` object at `site`.
+    If `site==len(self) - 1` `BaseMPS.connector_matrix`
+    is absorbed fromt the right-hand side into the returned 
+    `Node` object.
+
+    Args:
+      site: The site for which to return the `Node`.
+    Returns:
+      `Node`: The node at `site`.
+    """
+    if site >= len(self):
+      raise IndexError(
+          'index `site` = {} is out of range for len(mps)= {}'.format(
+              site, len(self)))
+    if site < 0:
+      raise ValueError(
+          'index `site` has to be larger than 0 (found `site`={}).'.format(
+              site))
+    if (site == len(self) - 1) and (self.connector_matrix is not None):
+      self.nodes[site][2] ^ self.connector_matrix[0]
+      order = [
+          self.nodes[site][0], self.nodes[site][1], self.connector_matrix[1]
+      ]
+      return contract_between(
+          self.nodes[site],
+          self.connector_matrix,
+          name=self.nodes[site].name,
+          output_edge_order=order)
+    return self.nodes[site]
+
+  def canonicalize(self, normalize: Optional[bool] = True) -> np.number:
+    raise NotImplementedError()
+
+
+class InfiniteMPS(BaseMPS):
+  """
+  An MPS class for infinite systems. 
+
+  MPS tensors are stored as a list of `Node` objects in the `InfiniteMPS.nodes`
+  attribute.
+  `InfiniteMPS` has a central site, also called orthogonality center. 
+  The position of this central site is stored in `InfiniteMPS.center_position`, 
+  and it can be be shifted using the `InfiniteMPS.position` method. 
+  `InfiniteMPS.position` uses QR and RQ methods to shift `center_position`.
+  
+  `InfiniteMPS` can be initialized either from a `list` of tensors, or
+  by calling the classmethod `InfiniteMPS.random`.
+  """
+
+  def __init__(self,
+               tensors: List[Union[BaseNode, Tensor]],
+               center_position: Optional[int] = 0,
+               connector_matrix: Optional[Union[BaseNode, Tensor]] = None,
+               backend: Optional[Text] = None) -> None:
+    """
+    Initialize a FiniteMPS.
+    Args:
+      tensors: A list of `Tensor` or `BaseNode` objects.
+      center_position: The initial position of the center site.
+      connector_matrix: A `Tensor` or `BaseNode` of rank 2 connecting
+        different unitcells. A value `None` is equivalent to an identity
+        `connector_matrix`.
+      backend: The name of the backend that should be used to perform 
+        contractions. Available backends are currently 'numpy', 'tensorflow',
+        'pytorch', 'jax'
+    """
+
+    super().__init__(
+        tensors=tensors,
+        center_position=center_position,
+        connector_matrix=connector_matrix,
+        backend=backend)
+
+  @classmethod
+  def random(cls,
+             d: List[int],
+             D: List[int],
+             dtype: Type[np.number],
+             backend: Optional[Text] = None):
```

Comment:
Optional: Perhaps also allow backend objects in addition to backend names.

Reply:
good point, I'll add that

Reply:
it's already supported, changed typing